### PR TITLE
Feature/rx java support

### PR DESCRIPTION
**Pull request by stefanmedack**
This PR encapsulates the current code into an own module `base` and adds an additional module `rx-java-2`

### CODE REVIEW

File: rx-java-2/gradle.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+module_artifact_id=rx-java-2
```
Comment (johnjohndoe) [inline rx-java-2/gradle.properties]: Please use camel-case here: `moduleArtifactId`.

File: base/gradle.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+module_artifact_id=base
```
Comment (johnjohndoe) [inline base/gradle.properties]: Please use camel-case here: `moduleArtifactId`.

File: README.md [hunk 3]
```diff
@@ -32,7 +32,10 @@ and to your application module `build.gradle`:
 
 ```groovy
 dependencies {
-    compile "info.metadude.kotlin.library.c3media:c3media-base:${version}"
+    // use this for the basic c3media version
+    compile "info.metadude.kotlin.library.c3media:base:${version}"
```
Comment (johnjohndoe) [inline README.md]: Please remove curly braces around the inline variable since they are actually not needed.

**Review by johnjohndoe — CHANGES_REQUESTED** [review]
Thank you for this nice feature! :rocket: 

I like you to clean up your branch a bit.
1. Please **squash** the first three commits into **one** since all that happened was a name change of the module. We don't need to keep the history of this decision change.
1. Futher, rebase the last commit into the former to undo changing the **library version** back and forth. I will raise the version as soon as it is suitable.
1. Other changes needed I commented directly.

Please let me know if you need help with rebasing your branches - I can also do that otherwise. Feel free to **force push your branch** to update this pull request.

**Comment by stefanmedack** [comment]
Okay, changed everything you asked for. Feel free to have another look

File: base/gradle.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+moduleArtifactId=base
```
Comment (johnjohndoe) [inline base/gradle.properties:1]: Please prefix the artifact id with `c3media-` as it was before. The name of the generated jar file should always state the name of the library such as `c3media-base-0.1.0-SNAPSHOT.jar`. As a result you also have to update the dependency statement to `compile "info.metadude.kotlin.library.c3media:c3media-base:$version"`.

File: rx-java-2/gradle.properties [hunk 5]
```diff
@@ -0,0 +1 @@
+moduleArtifactId=rx-java-2
```
Comment (johnjohndoe) [inline rx-java-2/gradle.properties:1]: Please prefix the artifact id with `c3media-` as it was before. The name of the generated jar file should always state the name of the library such as `c3media-rx-java-2-0.1.0-SNAPSHOT.jar`. As a result you also have to update the dependency statement to `compile "info.metadude.kotlin.library.c3media:c3media-rx-java-2:$version"`.

File: gradle/deploy.gradle [hunk 6]
```diff
@@ -9,7 +9,7 @@ publishing {
         mavenJava(MavenPublication) {
             from components.java
             groupId project.groupId
-            artifactId module_artifact_id
+            artifactId moduleArtifactId
```
Comment (johnjohndoe) [inline gradle/deploy.gradle]: Rebase interactive your changes into the original commits - don't put the changes on top of the branch.

File: README.md [hunk 7]
```diff
@@ -33,9 +33,9 @@ and to your application module `build.gradle`:
 ```groovy
 dependencies {
     // use this for the basic c3media version
-    compile "info.metadude.kotlin.library.c3media:base:${version}"
```
Comment (johnjohndoe) [inline README.md]: Rebase interactive your changes into the original commits - don't put the changes on top of the branch.

File: gradle.properties [hunk 8]
```diff
@@ -1,9 +1,8 @@
 # Release information
-versionName=0.1.0-SNAPSHOT
+versionName=0.2.0-SNAPSHOT
```
Comment (johnjohndoe) [inline gradle.properties]: Don't change the version here.

File: gradle.properties [hunk 9]
```diff
@@ -1,5 +1,5 @@
 # Release information
-versionName=0.4.0-SNAPSHOT
+versionName=0.1.0-SNAPSHOT
```
Comment (johnjohndoe) [inline gradle.properties]: Don't change the version here.

File: gradle.properties [hunk 10]
```diff
@@ -1,5 +1,5 @@
 # Release information
-versionName=0.2.0-SNAPSHOT
+versionName=0.4.0-SNAPSHOT
```
Comment (johnjohndoe) [inline gradle.properties]: Don't change the version here.

**Review by johnjohndoe — CHANGES_REQUESTED** [review]
Thanks for the corrections. Please use `git rebase -i` to clean up your branch. The later corrections should not sit on top of the branch instead should be integrated in the commits which introduced the code change.

I added a few new change requests.

File: base/gradle.properties (hunk 4 above)
Comment (johnjohndoe) [reply base/gradle.properties:1]: Fixed in #2.

File: rx-java-2/gradle.properties (hunk 5 above)
Comment (johnjohndoe) [reply rx-java-2/gradle.properties:1]: Fixed in #2.

File: gradle/deploy.gradle (hunk 6 above)
Comment (johnjohndoe) [reply gradle/deploy.gradle]: Fixed in #2.

**Pull request closed without merging**